src/utils/http/hyper_helpers.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any
+// person obtaining a copy of this software and associated
+// documentation files (the "Software"), to deal in the
+// Software without restriction, including without
+// limitation the rights to use, copy, modify, merge,
+// publish, distribute, sublicense, and/or sell copies of
+// the Software, and to permit persons to whom the Software
+// is furnished to do so, subject to the following
+// conditions:
+//
+// The above copyright notice and this permission notice
+// shall be included in all copies or substantial portions
+// of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+// ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT
+// SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+// CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR
+// IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+//! Utility methods relying on hyper
+
+use crate::types::error::GenericTransportError;
+use crate::types::http::HttpConfig;
+use futures::StreamExt;
+
+/// Read a hyper response with configured `HTTP` settings.
+///
+/// Returns `Ok(bytes)` if the body was in valid size range.
+/// Returns `Err` if the body was too large or the body couldn't be read.
+pub async fn read_response_to_body(
+	headers: &hyper::HeaderMap,
+	mut body: hyper::Body,
+	config: HttpConfig,
+) -> Result<Vec<u8>, GenericTransportError<hyper::Error>> {
+	// NOTE(niklasad1): Bigger values than `u32::MAX` will be regarded as zero here and that's very unlikely to occur.
+	// Thus, in those causes we rely on the while loop to allocate instead of `pre-allocate`.
```

Comment:
```suggestion
	// NOTE(niklasad1): Values bigger than `u32::MAX` will be turned into zero here. This is unlikely to occur in practise
	// and for that case we fallback to allocating in the while-loop below instead of pre-allocating.
```